apps/climatemappedafrica/src/components/Hero/Map.js

```diff
@@ -0,0 +1,134 @@
+import makeStyles from "@mui/styles/makeStyles";
+import { useRouter } from "next/router";
+import PropTypes from "prop-types";
+import React from "react";
+import { MapContainer, GeoJSON } from "react-leaflet";
+
+import "leaflet/dist/leaflet.css";
+import theme from "@/climatemappedafrica/theme";
+
+const useStyles = makeStyles(({ breakpoints, typography }) => ({
+  root: {
+    position: "relative",
+    height: typography.pxToRem(299),
+    width: typography.pxToRem(236),
+    marginTop: typography.pxToRem(55),
+    [breakpoints.up("lg")]: {
+      height: typography.pxToRem(471),
+      marginTop: typography.pxToRem(42),
+      width: typography.pxToRem(371),
+    },
+    "& .leaflet-container": {
+      background: "transparent",
+    },
+  },
+  tooltip: {
+    fontFamily: typography.body1.fontFamily,
+    fontSize: typography.pxToRem(13),
+    color: "#2A2A2C",
+    textTransform: "capitalize",
+  },
+}));
+
+function Map({
+  center,
+  zoom,
+  boundary,
+  styles = {
+    height: "100%",
+    width: "100%",
+  },
+  geoJSONStyles = {
+    color: "#2A2A2C",
+    weight: 1,
+    opacity: 1,
+    fillColor: "#fff",
+    dashArray: "2",
+  },
+  setHoverGeo,
+  featuredLocations,
+  ...props
+}) {
+  const classes = useStyles(props);
+  const router = useRouter();
+
+  const countyCodes = featuredLocations?.map(({ code }) => code);
+
+  const onEachFeature = (feature, layer) => {
+    layer.setStyle({
+      fillColor: theme.palette.background.default,
+      fillOpacity: 1,
+    });
+
+    if (countyCodes.includes(feature.properties.code?.toLowerCase())) {
+      layer.setStyle({
+        weight: 1.5,
+        dashArray: 0,
+      });
+      layer.on("mouseover", () => {
+        setHoverGeo(feature.properties.name.toLowerCase());
+        layer.setStyle({
+          fillColor: theme.palette.primary.main,
+          fillOpacity: 0.5,
+        });
+      });
+      layer.on("mouseout", () => {
+        setHoverGeo(null);
+        layer.setStyle({
+          fillOpacity: 1,
+          fillColor: theme.palette.background.default,
+        });
+      });
+      layer.on("click", () => {
+        router.push(`/explore/${feature.properties.code.toLowerCase()}`);
```

Comment:
This should not be hardcoded. It will be fixed with https://github.com/CodeForAfrica/ui/pull/945. We can add an issue to track it later